CMakeLists.txt

```diff
@@ -987,6 +998,10 @@ foreach(hal ${OpenCV_HAL})
     else()
       message(STATUS "Carotene: NEON is not available, disabling carotene...")
     endif()
+  elseif(hal STREQUAL "kleidicv")
+    add_subdirectory(3rdparty/kleidicv)
```

Comment:
This is for OpenCV's 3rdparty "builtin" optimizations only.

Added case is external library and we should use here generic approach without specific implementation details).

1. See on the "else()" branch which uses `find_package()` - library should provide CMake's scripts for detection / usage.
2. Also we could add "PATH" support (e.g. through `if (IS_DIRECTORY "${hal}")` check)

Reply:
Can KleidiCV be integrated via [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) please?

In that way it can be treated more like the other 3rdparty HALs, and can be enabled in OpenCV with a simple `-DWITH_KLEIDICV=ON` cmake argument.